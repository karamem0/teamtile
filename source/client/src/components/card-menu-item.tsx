//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Text } from '@fluentui/react-northstar';

import { css } from '@emotion/react';

export interface CardMenuItemProps {
  icon?: React.ReactNode,
  content?: React.ReactNode,
  onClick?: React.MouseEventHandler
}

export const CardMenuItem = ({
  icon,
  content,
  onClick
}: CardMenuItemProps): React.ReactElement | null => {

  return (
    <Text
      color="brand"
      css={css`
        display: grid;
        grid-template-columns: auto auto;
        grid-template-rows: auto;
        grid-gap: 0.25rem;
        align-items: flex-end;
        justify-content: flex-end;
        cursor: pointer;
      `}
      role="button"
      onClick={onClick}>
      {
        icon && (
          <span
            css={css`
            line-height: 1rem;
            width: 1rem;
            height: 1rem;
          `}>
            {icon}
          </span>
        )
      }
      {
        content && (
          <Text
            content={content}
            css={css`
              line-height: 0.75rem;
            `}
            size="small" />
        )
      }
    </Text>
  );

};
