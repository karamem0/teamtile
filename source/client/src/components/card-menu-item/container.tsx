//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { EventHandler } from '../../types/common';

import Presenter from './presenter';

interface CardMenuItemProps {
  icon: React.ReactNode | undefined,
  content: React.ReactNode | undefined,
  onClick?: EventHandler | undefined
}

export default function CardMenuItem ({
  icon,
  content,
  onClick
}: CardMenuItemProps): React.ReactElement | null {

  return (
    <Presenter
      content={content}
      icon={icon}
      onClick={onClick} />
  );

}
