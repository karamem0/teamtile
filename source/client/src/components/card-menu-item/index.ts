//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import Container from './container';
export {
  Container as CardMenuItem
};
