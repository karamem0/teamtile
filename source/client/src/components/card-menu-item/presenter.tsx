//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Text } from '@fluentui/react-northstar';

import { css } from '@emotion/react';

import { EventHandler } from '../../types/common';

interface CardMenuItemProps {
  icon: React.ReactNode | undefined,
  content: React.ReactNode | undefined,
  onClick?: EventHandler | undefined
}

export default React.memo(function CardMenuItem ({
  icon,
  content,
  onClick
}: CardMenuItemProps): React.ReactElement | null {

  return (
    <Text
      color="brand"
      css={css`
        display: grid;
        grid-template-rows: auto;
        grid-template-columns: auto auto;
        grid-gap: 0.25rem;
        align-items: flex-end;
        justify-content: flex-end;
        cursor: pointer;
      `}
      role="button"
      onClick={(event: React.SyntheticEvent) => onClick && onClick(event)}>
      {
        icon &&
        (
          <span
            css={css`
            width: 1rem;
            height: 1rem;
            line-height: 1rem;
          `}>
            {icon}
          </span>
        )
      }
      <Text
        content={content}
        css={css`
          line-height: 0.75rem;
        `}
        size="small" />
    </Text>
  );

});
