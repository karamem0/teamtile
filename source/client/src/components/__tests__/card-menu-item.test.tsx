//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import { CardMenuItem } from '../card-menu-item';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('CardMenuItem', () => {

  it('create shapshot', () => {
    const params = {
      icon: 'icon',
      content: 'content',
      onClick: jest.fn()
    };
    render(<CardMenuItem {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
