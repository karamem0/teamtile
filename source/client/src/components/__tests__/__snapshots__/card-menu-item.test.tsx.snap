// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMenuItem create shapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: auto auto;
  grid-template-rows: auto;
  grid-gap: 0.25rem;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  cursor: pointer;
}

.emotion-1 {
  line-height: 1rem;
  width: 1rem;
  height: 1rem;
}

.emotion-2 {
  line-height: 0.75rem;
}

<body>
  <div>
    <span
      class="ui-text emotion-0"
      data-uses-unhanded-props="true"
      role="button"
    >
      <span
        class="emotion-1"
      >
        icon
      </span>
      <span
        class="ui-text emotion-2"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        content
      </span>
    </span>
  </div>
</body>
`;
